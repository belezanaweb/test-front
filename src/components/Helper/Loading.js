import React from 'react'
import { RollerSpinner } from './styles'

const Loading = ({ color }) => (
  <RollerSpinner color={color} data-testid="loading">
    <div />
    <div />
    <div />
    <div />
    <div />
    <div />
    <div />
    <div />
  </RollerSpinner>
)

Loading.defaultProps = {
  size: 64,
  color: '#FF7800'
}

export default Loading
