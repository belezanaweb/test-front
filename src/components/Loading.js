import React from 'react';
import styled from 'styled-components';

const LoadingContainer = styled.div`
  display: flex;
  justify-content: center;
  align-items: center;
  height: calc(100vh - 6rem);
  color: var(--color-orange-1);
`

const Loading = () => (
  <LoadingContainer>
    <i className='fa fa-spinner fa-pulse fa-5x' aria-hidden="true"></i>
  </LoadingContainer>
)

export default Loading
