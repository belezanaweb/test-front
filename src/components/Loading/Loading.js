import React from 'react'
import { keyframes } from '@emotion/core'
import styled from '@emotion/styled/macro'
import logo from './logo.svg'

const LoadingContainer = styled.div`
  align-items: center;
  background-color: #282c34;
  color: white;
  display: flex;
  flex-direction: column;
  font-size: calc(10px + 2vmin);
  justify-content: center;
  min-height: 100vh;
  text-align: center;
`

const logoSpin = keyframes`
  from {
    transform: rotate(0deg);
  }
  to {
    transform: rotate(360deg);
  }
`

const LoadingImage = styled.img`
  animation: ${logoSpin} infinite 20s linear;
  height: 40vmin;
  pointer-events: none;
`

function Loading () {
  return (
    <LoadingContainer>
      <LoadingImage src={logo} alt="logo" />
    </LoadingContainer>
  )
}

export default Loading
