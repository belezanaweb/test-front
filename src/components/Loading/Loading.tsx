import React from 'react'

export const Loading = () => <h1>Loading...</h1>
