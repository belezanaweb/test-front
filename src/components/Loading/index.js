import React from 'react';
import { makeStyles } from '@material-ui/core/styles';
import CircularProgress from '@material-ui/core/CircularProgress';
import { Container } from './styles';

const useStyles = makeStyles((theme) => ({
    loading: {
        color: '#ffffff',
    },
}));

export default function Loading() {
    const classes = useStyles();

    return (
        <Container>
            <CircularProgress className={classes.loading} />
        </Container>
    );
};