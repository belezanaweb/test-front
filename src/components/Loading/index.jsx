import React from 'react'
import { LoadingBox, Icon } from './styled'
import preloader from './images/preloader.svg'

export default function Loading() {
  return (
    <LoadingBox>
      <Icon src={preloader} alt="Loading" />
    </LoadingBox>
  )
}
