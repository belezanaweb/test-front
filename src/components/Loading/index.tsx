import { Container } from './styles';

export default function Loading() {
  return (
    <Container>
      <div className="lds-ring">
        <div></div>
        <div></div>
        <div></div>
        <div></div>
      </div>
    </Container>
  );
}
