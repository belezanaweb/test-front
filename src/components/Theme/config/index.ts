export { default as light } from "./light";
