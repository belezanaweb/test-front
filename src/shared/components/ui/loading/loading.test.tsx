import React from 'react';
import { render, screen } from '../../../../../config/test-utils';

import Loading from './index';

describe('<Loading />', () => {
  it('should render component with <div> tag', () => {
    render(<Loading />, {});
    expect(screen.findByRole('div'));
  })

  it('should render component with default props and match snapshot', () => {
    const { container } = render(<Loading />, {});
    expect(container.firstChild).toMatchSnapshot();
  })
})
