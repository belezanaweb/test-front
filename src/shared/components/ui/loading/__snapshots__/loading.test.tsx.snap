// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Loading /> should render component with default props and match snapshot 1`] = `
.c0 {
  width: 30px;
  height: 30px;
  position: relative;
  overflow: hidden;
  -webkit-animation: hWuqhH 2.4s linear infinite;
  animation: hWuqhH 2.4s linear infinite;
}

.c0 > div {
  width: 12px;
  height: 12px;
  border-radius: 50%;
  position: absolute;
  background-color: #ff6c00;
  -webkit-animation: dfNaGE 2.4s linear infinite alternate;
  animation: dfNaGE 2.4s linear infinite alternate;
}

.c0 > div:nth-of-type(1) {
  -webkit-animation-delay: 0s;
  animation-delay: 0s;
}

.c0 > div:nth-of-type(2) {
  -webkit-animation-delay: 0.7999999999999999s;
  animation-delay: 0.7999999999999999s;
}

.c0 > div:nth-of-type(3) {
  -webkit-animation-delay: 1.5999999999999999s;
  animation-delay: 1.5999999999999999s;
}

.c0 > div:nth-of-type(4) {
  -webkit-animation-delay: 2.4s;
  animation-delay: 2.4s;
}

.c1 {
  left: 0;
  top: 0;
}

.c2 {
  right: 0;
  top: 0;
}

.c3 {
  left: 0;
  bottom: 0;
}

.c4 {
  right: 0;
  bottom: 0;
}

<div
  class="c0"
  color="#ff6c00"
  speed="2.4"
  style="margin: auto; position: absolute; left: 0px; right: 0px; top: 0px; bottom: 0px;"
>
  <div
    class="c1"
  />
  <div
    class="c2"
  />
  <div
    class="c3"
  />
  <div
    class="c4"
  />
</div>
`;
